corehq/apps/sms/models.py

```diff
@@ -272,6 +273,15 @@ def outbound_backend(self):
             domain=self.domain
         )
 
+    def set_status_pending(self):
+        """Mark message as sent with backend status pending"""
+        self.error = False
+        self.system_error_message = SMSBase.STATUS_PENDING
```

Comment:
Indeed that is true, and it is not ideal. I do not love this, but the SMS model does not appear to have any other way of encoding this type of information. I also set `self.error = False` so that it is clear this is not an error, and the only other place that uses `SMSBase.STATUS_PENDING` is immediately below (in `is_status_pending`), so it is fully encapsulated in this class. The "right way" to do it might be to add another field to this class to hold status information, but that seems overkill to meet the needs of one gateway that does things differently from all of our other gateway adapters.

Reply:
> I also set self.error = False so that it is clear this is not an error

That is a fair point.
Just that from SMS model perspective, it is independent of the gateway, so it causes a confusion there like someone could also read it as that, the SMS had an error at certain point and then later was sent on a retry (or may be not if we clean the system_error_message if we set error=False but then we are doing the opposite here). This is just creates more branches that might not be easy to see.

Reply:
They are both reset if the message is requeued
https://github.com/dimagi/commcare-hq/blob/ead9f50055c26769afa81ae44cdbab2c1322a3cb/corehq/apps/sms/models.py#L312-L313